File: src/test/java/com/github/kayjamlang/tests/containers/classes/ClassImplementsContainerTest.java
package com.github.kayjamlang.tests.containers.classes;

import com.github.kayjamlang.core.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.ClassContainer;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ClassImplementsContainerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("class Test:: ABC {var test = 123;fun test(){return test}}"));
    }

    @Test
    public void test() throws Exception {
        Expression expression = parser.readExpression();

        assertNotNull(expression);
        assertSame(ClassContainer.class, expression.getClass());

        ClassContainer classContainer = (ClassContainer) expression;
        assertEquals("Test", classContainer.name);
        assertEquals(1, classContainer.implementsClass.size());
        assertEquals("ABC", classContainer.implementsClass.get(0));
        assertEquals(1, classContainer.children.size());
        assertEquals(1, classContainer.functions.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/classes/ClassImplementsContainerTest.scala
package com.github.kayjamlang.tests.containers.classes

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.ClassContainer
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ClassImplementsContainerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer("class Test:: ABC {var test = 123;fun test(){return test}}"))
  }
}

class ClassImplementsContainerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ClassImplementsContainerTest.parser.readTopExpression
    assertNotNull(expression)
    assertSame(classOf[ClassContainer], expression.getClass)
    val classContainer = expression.asInstanceOf[ClassContainer]
    assertEquals("Test", classContainer.name)
    assertEquals(1, classContainer.implementsClass.size)
    assertEquals("ABC", classContainer.implementsClass.apply(0))
    assertEquals(1, classContainer.variables.size)
    assertEquals(1, classContainer.functions.size)
  }
}